mixer/adapter/stackdriver/trace/trace.go

```diff
@@ -42,6 +43,25 @@ type (
 	}
 )
 
+// NOTE: because Istio 1.0.X branch uses 64bit traceids, it is safe to use this sampling
+// logic for the adapter.
+func sixtyFourBitTraceIDSampler(fraction float64) trace.Sampler {
+	if !(fraction >= 0) {
+		fraction = 0
+	} else if fraction >= 1 {
+		return trace.AlwaysSample()
+	}
+
+	traceIDUpperBound := uint64(fraction * (1 << 63))
+	return trace.Sampler(func(p trace.SamplingParameters) trace.SamplingDecision {
+		if p.ParentContext.IsSampled() {
+			return trace.SamplingDecision{Sample: true}
+		}
+		x := binary.BigEndian.Uint64(p.TraceID[8:])
```

Comment:
I think this needs to be shift 1 bit to the right for comparison: https://github.com/census-instrumentation/opencensus-go/blob/master/trace/sampling.go#L55

Reply:
I, too, was initially under that impression based on the source.  But that's why I added the `TestSampler` test.  If we shift right, we get double the expected samples for any specified rate (as we are halving the sample values).

Reply:
Whoops. My test code was bad. Fixed.